loadbalancer/algorithm.go

```diff
@@ -101,14 +117,15 @@ func withFadeIn(rnd *rand.Rand, ctx *routing.LBContext, w []int, choice int) rou
 		return ctx.Route.LBEndpoints[choice]
 	}
 
-	return shiftToRemaining(rnd, ctx, w, now)
+	return shiftToRemaining(rnd, ctx, wi, wf, now)
 }
 
 type roundRobin struct {
-	mx         sync.Mutex
-	index      int
-	rnd        *rand.Rand
-	fadeInCalc []int
+	mx               sync.Mutex
+	index            int
+	rnd              *rand.Rand
+	fadeInCalcInts   []int
+	fadeInCalcFloats []float64
```

Comment:
fadeInCalcInts and fadeInCalcFloats are not obvious names. 
fadeInCalcInts is stored in notFadingIndexes, which is much more clear IMO. 
And fadeInCalcFloats seem to be fadingWeights or fadingPercentage.

wdyt?

Reply:
sounds good to me, thx! Might happen that earlier the ints were used for different purposes as worker slices, but now there is only one.